dynamicio/io/keyed.py

```diff
@@ -0,0 +1,70 @@
+# pylint: disable=W0707
+"""KeyedResource class for reading and writing to different resources based on a key."""
+from __future__ import annotations
+
+from copy import deepcopy
+from typing import Protocol, Tuple, Type
+
+import pandas as pd
+from pandera import SchemaModel
+
+
+class IOConfig(Protocol):
```

Comment:
Very Pythonic usage of protocols instead of `ABC` metaclasses!

Reply:
Yeah, imo we should be using ABC. XD